src/openrct2-ui/windows/StaffFirePrompt.cpp

```diff
@@ -39,71 +43,57 @@ static rct_widget window_staff_fire_widgets[] = {
     WIDGETS_END,
 };
 
-static void WindowStaffFireMouseup(rct_window *w, WidgetIndex widgetIndex);
-static void WindowStaffFirePaint(rct_window *w, rct_drawpixelinfo *dpi);
-
-//0x9A3F7C
-static WindowEventList window_staff_fire_events([](auto& events)
-{
-    events.mouse_up = &WindowStaffFireMouseup;
-    events.paint = &WindowStaffFirePaint;
-});
 // clang-format on
 
-/** Based off of rct2: 0x6C0A77 */
-rct_window* WindowStaffFirePromptOpen(Peep* peep)
+class StaffFirePromptWindow final : public Window
 {
-    rct_window* w;
-
-    // Check if the confirm window already exists.
-    w = window_bring_to_front_by_number(WindowClass::FirePrompt, peep->sprite_index.ToUnderlying());
-    if (w != nullptr)
+public:
+    void Init(rct_windownumber windownumber)
```

Comment:
I think `SetWindowNumber` or similar would sound better.

Reply:
Function name changed to `SetWindowNumber`